pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2188,14 +2198,20 @@ func DomainCreate(status types.DomainStatus) (int, string, error) {
 		// Use rkt tool
 		// get the rkt image hash for this file; if we do not have it in the rkt cache,
 		// convert it
-		// XXX we assume a container has one image. XXX do we check somewhere?
-		ociFilename, err := utils.VerifiedImageFileLocation(status.DiskStatusList[0].ImageSha256)
+		var containerImageSha256 string
+		for _, ds := range status.DiskStatusList {
+			if ds.Format == zconfig.Format_CONTAINER {
+				containerImageSha256 = ds.ImageSha256
+				break
+			}
+		}
+		ociFilename, err := utils.VerifiedImageFileLocation(containerImageSha256)
```

Comment:
Don't you also need to check that numOfContainerDisks == 0 somewhere and return an error?  otherwise you will have no containerImageSha256 here.

Reply:
Yes, we should add check for numOfContainerDisks == 0. I will update the PR.

Reply:
@nordmark I checked it if there is no container disk in the storage list, we are handling the instance as VM not container. So, as per our current implementation we don't need this check. We are deriving instance type (is container or not) by looking at its storage list.
For reference: https://github.com/lf-edge/eve/blob/2f627ba5d2dd6b48398c7de2b883b5d11c27096a/pkg/pillar/cmd/zedmanager/zedmanager.go#L562